Source/Charts/Utils/ChartColorTemplates.swift

```diff
@@ -129,19 +129,39 @@ open class ChartColorTemplates: NSObject
         else if colorString.hasPrefix("rgba")
         {
             var a: Float = 1.0
-            var r: Int32 = 0
-            var g: Int32 = 0
-            var b: Int32 = 0
-            let scanner: Scanner = Scanner(string: colorString)
-            scanner.scanString("rgba", into: nil)
-            scanner.scanCharacters(from: leftParenCharset, into: nil)
-            scanner.scanInt32(&r)
-            scanner.scanCharacters(from: commaCharset, into: nil)
-            scanner.scanInt32(&g)
-            scanner.scanCharacters(from: commaCharset, into: nil)
-            scanner.scanInt32(&b)
-            scanner.scanCharacters(from: commaCharset, into: nil)
-            scanner.scanFloat(&a)
+            var r: Int64 = 0
```

Comment:
Eventually (when only supporting iOS 13+ we will be removing the alternative implementation. Rather than try to match the old implementation, it would be better to use a new implementation that will allow us to just delete the old code altogether without having to think about what we are deleting. It's also a good opportunity to improve the implementation. One way to do this, which I think feels a little better:
```
let scanner: Scanner = Scanner(string: colorString)
scanner.charactersToBeSkipped = commaCharset.union(leftParenCharset)
if #available(iOS 13.0, *)
{
    _ = scanner.scanString("rgba")
    let r = scanner.scanInt() ?? 0
    let g = scanner.scanInt() ?? 0
    let b = scanner.scanInt() ?? 0
    let a = scanner.scanFloat(representation: .decimal) ?? 1.0

    return NSUIColor(
        red: CGFloat(r) / 255.0,
        green: CGFloat(g) / 255.0,
        blue: CGFloat(b) / 255.0,
        alpha: CGFloat(a)
    )
}
else
{
    var a: Float = 1.0
    var r: Int32 = 0
    var g: Int32 = 0
    var b: Int32 = 0

    scanner.scanString("rgba", into: nil)
    scanner.scanInt32(&r)
    scanner.scanInt32(&g)
    scanner.scanInt32(&b)
    scanner.scanFloat(&a)

    return NSUIColor(
        red: CGFloat(r) / 255.0,
        green: CGFloat(g) / 255.0,
        blue: CGFloat(b) / 255.0,
        alpha: CGFloat(a)
    )
}
```

Reply:
Forgot to show:
```
let scanner = Scanner(string: colorString)
scanner.charactersToBeSkipped = commaCharset.union(leftParenCharset)
```